File: backend/internal/services/settings_service.go
package services

import (
	"trinity-lodge/internal/models"
	"trinity-lodge/internal/repository"

	"github.com/google/uuid"
)

type SettingsService struct {
	repo *repository.SettingsRepository
}

func NewSettingsService(repo *repository.SettingsRepository) *SettingsService {
	return &SettingsService{repo: repo}
}

func (s *SettingsService) Get(userID uuid.UUID) (*models.Settings, error) {
	settings, err := s.repo.Get(userID)
	if err != nil {
		// Return default settings if not found
		return &models.Settings{
			UserID:                  userID,
			LodgeName:               "My Lodge",
			Address:                 "",
			Phone:                   "",
			GSTNumber:               "",
			StateName:               "",
			StateCode:               "",
			GSTInvoicePrefix:        "GST",
			GSTInvoiceNextNumber:    1,
			NonGSTInvoicePrefix:     "INV",
			NonGSTInvoiceNextNumber: 1,
		}, nil
	}
	return settings, nil
}

func (s *SettingsService) Save(settings *models.Settings, userID uuid.UUID) error {
	return s.repo.Upsert(settings, userID)
}

func (s *SettingsService) CreateDefaultSettings(userID uuid.UUID) error {
	return s.repo.CreateDefaultSettings(userID)
}

File: frontend/src/services/settings.service.ts
// frontend/src/services/settings.service.ts
import { SettingsAPI } from '@/lib/bindings'
import type { Settings } from '@/types'

export const settingsService = {
  get: () => SettingsAPI.Get() as unknown as Promise<Settings>,
  save: (s: Omit<Settings, 'id' | 'created_at' | 'updated_at'>) =>
    SettingsAPI.Save({
      lodge_name: s.lodge_name, address: s.address, phone: s.phone,
      gst_number: s.gst_number, state_name: s.state_name, state_code: s.state_code,
      gst_invoice_prefix: s.gst_invoice_prefix,
      gst_invoice_next_number: s.gst_invoice_next_number,
      non_gst_invoice_prefix: s.non_gst_invoice_prefix,
      non_gst_invoice_next_number: s.non_gst_invoice_next_number,
    }) as unknown as Promise<Settings>,
}
